src/Core/src/Handlers/Application/ApplicationHandler.Windows.cs

```diff
@@ -19,7 +19,20 @@ public partial class ApplicationHandler : ElementHandler<IApplication, UI.Xaml.A
 		{
 			if (args is IWindow window)
 			{
-				(window.Handler?.PlatformView as Window)?.Close();
+				var platformWindow = window.Handler?.PlatformView as Window;
+
+				if (platformWindow is not null)
+				{
+					var handle = WinRT.Interop.WindowNative.GetWindowHandle(platformWindow);
+					var id = UI.Win32Interop.GetWindowIdFromWindow(handle);
+					var appWindow = UI.Windowing.AppWindow.GetFromWindowId(id);
+
+					// Cannot close an already closed, or disposed Window.
+					if (appWindow is not null && appWindow.IsVisible)
```

Comment:
I think this is fine. `IsVisible` maps to window style of `WS_VISIBLE`. I was worried that minimized windows might not have that set, but docs say min/restore windows doesn't change that at all :)

Reply:
What if the window is hidden? We can't ever close?

You can make windows invisible on Windows for a few reasons - like making a screen clipping app.